### PR TITLE
feat: azure linux 3 support

### LINKED ISSUE #697: [REQ] add tests for azure linux 3
### What kind of request is this?

Other

### What is your request or suggestion?

adding tests for azure linux 3 preview core and distroless images 

### Are you willing to submit PRs to contribute to this feature request?

- [ ] Yes, I am willing to implement it.

### PR DESCRIPTION
**Pull request by sozercan**
Describe the changes in this pull request using active verbs such as _Add_, _Remove_, _Replace_ ...

Add Azure Linux 3 support and tests

Closes #697

### CODE REVIEW

File: pkg/pkgmgr/rpm.go [hunk 1]
```diff
@@ -107,19 +108,26 @@ func isLessThanRPMVersion(v1, v2 string) bool {
...
-		vers := strings.Split(manifest.Metadata.OS.Version, ".")
-		if len(vers) < 2 {
-			vers = append(vers, "0")
-		}
-		version = fmt.Sprintf("%s.%s", vers[0], vers[1])
+func getRPMImageName(manifest *unversioned.UpdateManifest, osType string, osVersion string) string {
+	var image, version string
+
+	if osType == "azurelinux" {
+		image = "mcr.microsoft.com/azurelinux/base/core"
```
Comment (ashnamehrotra) [inline pkg/pkgmgr/rpm.go:115]: I think we should also update the FAQ tooling image section to reflect this tooling image for azurelinux

File: pkg/pkgmgr/rpm.go [hunk 2]
```diff
@@ -449,14 +457,18 @@ func (rm *rpmManager) installUpdates(ctx context.Context, updates unversioned.Up
 	// Install patches using available rpm managers in order of preference
 	var installCmd string
 	switch {
-	case rm.rpmTools["dnf"] != "":
-		checkUpdateTemplate := `sh -c "%[1]s check-update; if [ $? -ne 0 ]; then echo >> /updates.txt; fi"`
-		if !rm.checkForUpgrades(ctx, rm.rpmTools["dnf"], checkUpdateTemplate) {
+	case rm.rpmTools["tdnf"] != "" || rm.rpmTools["dnf"] != "":
+		dnfTooling := rm.rpmTools["tdnf"]
+		if dnfTooling == "" {
```
Comment (ashnamehrotra) [inline pkg/pkgmgr/rpm.go:462]: what is this check case for?

Comment (sozercan) [reply pkg/pkgmgr/rpm.go:462]: existing code didn't account for tdnf. AL3 doesn't have dnf or yum, so it was failing to find the proper tooling. in this case, we are checking if tdnf is present. if not, falling back to dnf.

File: pkg/pkgmgr/rpm.go (hunk 1 above)
Comment (sozercan) [reply pkg/pkgmgr/rpm.go:115]: updated

**Review by ashnamehrotra — APPROVED** [review]
LGTM

**Pull request merged**